apps/src/gamelab/AnimationPicker/index.jsx

```diff
@@ -1,15 +1,76 @@
+var _ = require('../../lodash');
 var AnimationPickerBody = require('./AnimationPickerBody.jsx');
 var Dialog = require('../../templates/DialogComponent.jsx');
+var HiddenUploader = require('../../assetManagement/HiddenUploader.jsx');
+var styles = require('./styles');
+var utils = require('../../utils');
+
+/** @enum {string} */
+var View = {
+  PICKER: 'PICKER',
+  UPLOAD_IN_PROGRESS: 'UPLOAD_IN_PROGRESS',
+  ERROR: 'ERROR'
+};
 
 var AnimationPicker = React.createClass({
   propTypes: {
-    handleClose: React.PropTypes.func.isRequired
+    channelId: React.PropTypes.string.isRequired,
+    onComplete: React.PropTypes.func.isRequired,
+    onCancel: React.PropTypes.func.isRequired,
+    typeFilter: React.PropTypes.string
+  },
+
+  getInitialState: function () {
+    return {view: View.PICKER};
+  },
+
+  onUploadClick: function () {
+    this.refs.uploader.openFileChooser();
+  },
+
+  onUploadStart: function (data) {
+    this.setState({
+      view: View.UPLOAD_IN_PROGRESS,
+      originalFileName: data.files[0].name
+    });
```

Comment:
In a world where we have redux, it may make sense to have a discussion about if/where we still use local state. I think it would be good if we had a clear store on when it was reasonable.